addon/components/bs-navbar.js

```diff
@@ -0,0 +1,87 @@
+import Ember from 'ember';
+import TypeClass from 'ember-bootstrap/mixins/type-class';
+import layout from '../templates/components/bs-navbar';
+
+/**
+ Component to generate [bootstrap navbars](http://getbootstrap.com/components/#navbar).
+
+ ### Usage
+
+ Uses the following components by a contextual reference:
+
+ - [Components.NavbarHeader](Components.NavbarHeader.html
+ - [Components.NavbarContent](Components.NavbarContent.html
+ - [Components.NavbarToggle](Components.NavbarToggle.html
+ - [Components.NavbarBrand](Components.NavbarBrand.html
+ - [Components.NavbarNav](Components.NavbarNav.html
+
+ and uses [Components.NavItem](Components.NavItem.html) directly.
+
+ ```hbs
+ {{#bs-navbar as |navbar|}}
+   {{#navbar.header}}
+     {{#navbar.toggle}}
+       <span class="sr-only">Toggle navigation</span>
+       <span class="icon-bar"></span>
+       <span class="icon-bar"></span>
+       <span class="icon-bar"></span>
+     {{/navbar.toggle}}
+     {{#navbar.brand}}Brand{{/navbar.brand}}
+   {{/navbar.header}}
+   {{#navbar.content}}
+     {{#navbar.nav}}
+       {{#bs-nav-item}}{{#link-to "alert"}}Alert{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "button"}}Buttons{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "dropdown"}}Dropdown{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "forms"}}Forms{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "accordion"}}Accordion{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "collapse"}}Collapse{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "modal"}}Modals{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "progress"}}Progress bars{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "navs"}}Navs{{/link-to}}{{/bs-nav-item}}
+       {{#bs-nav-item}}{{#link-to "navbars"}}Navbars{{/link-to}}{{/bs-nav-item}}
+     {{/navbar.nav}}
+   {{/navbar.content}}
+ {{/bs-navbar}}
+ ```
+
+ ### Navbar styles
+
+ The component supports the default bootstrap navbar styling options "default" and "inverse" through the `type`
+ property. Bootstrap navbars [do not currently support justified nav links](http://getbootstrap.com/components/#navbar-default),
+ so those are explicitly disallowed.
+
+ Other bootstrap navbar variations, such as forms, buttons, fixed to top, etc. can be supported through direct use of
```

Comment:
I think we should support the styling options of the `bs-navbar` itself through properties, rather than setting class names. Something like a `position` property maybe?